pkg/apis/infinispan/v1/types_util.go

```diff
@@ -525,9 +534,31 @@ func applyLabels(ispn *Infinispan, envvar, annotationName string) error {
 	return err
 }
 
-// HasCustomLibraries true if custom libraries are defined
-func (ispn *Infinispan) HasCustomLibraries() bool {
-	return ispn.Spec.Dependencies != nil && ispn.Spec.Dependencies.VolumeClaimName != ""
+// GetExternalDependencyType return external dependency type if defined
+func (ispn *Infinispan) GetExternalDependencyType() ExternalDependencyType {
```

Comment:
I think only allowing one dependency type is too limiting. Infinispan Server recursively searches directories in `/opt/infinispan/server/lib` for files, so it's possible for us to support both `Dependencies.VolumeClaimName` & `Dependencies.ExternalLibraries` at the same time by using different mount paths within `server/lib`.

Reply:
Additionally, aside from jars, the server supports plain files (properties, etc). Is this allowed in the PR ?

Reply:
> I think only allowing one dependency type is too limiting. Infinispan Server recursively searches directories in `/opt/infinispan/server/lib` for files, so it's possible for us to support both `Dependencies.VolumeClaimName` & `Dependencies.ExternalLibraries` at the same time by using different mount paths within `server/lib`.

Good point, @ryanemerson will add this implementation to support both option working in parallel.

Reply:
> Additionally, aside from jars, the server supports plain files (properties, etc). Is this allowed in the PR ?

Yes, any type of the file can be download in two options:
- inside archive (preferred for multiple files)
- if extension not archive or skipped (URL rewriting) will download as original file